app/src/main/java/com/training/recyclerviewdemo/MainActivity.java

```diff
@@ -50,9 +57,30 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onClick(View view) {
         // add to the top of the list
         int topPos = 0;
-        personList.add(topPos, new Person("Anthony Jones", 22, "M"));
-        personsAdapter.notifyItemInserted(topPos);
-        // scroll to the top of the list
-        recyclerView.scrollToPosition(topPos);
+        String name =txtAddName.getText().toString();
+        int age;
+        String gender = txtAddGender.getText().toString();
+        String regExString = "^[0-9]*$";
+
```

Comment:
be generally mindful of spacing. keep spaces between '=', "+", "&&", and so forth